Samples/Salesforce.Sample.SmartSyncExplorerHybrid/typescript/salesforce.windows.smartsync.js

```diff
@@ -47,22 +47,18 @@ var SmartSyncJS;
             return this.smartSync.SyncManager.getInstance();
         };
         SmartSync.prototype.syncDown = function (success, fail, args) {
-            if (this.checkFirstArg(args))
-                return;
-            var payload = args[1];
             var syncmanager = this.getInstance();
             if (!syncmanager) {
                 fail("Error in getting instance for SmartSync");
             }
             else {
-                var syncState = syncmanager.syncDown(payload.target.asJson(), payload.soupName, null, payload.options);
+                var syncState = syncmanager.syncDown(args[0], args[1], args[2], args[3]);
                 success(syncState);
+                fail("Error in Sync Down");
```

Comment:
will cal success ad fail both, or should based the real status to decide to call success or fail?